sympy/polys/polyroots.py

```diff
@@ -504,33 +504,42 @@ def roots_cyclotomic(f, factor=False):
 
 def roots_quintic(f):
     """
-    Calculate exact roots of a solvable quintic
+    Calculate exact roots of a solvable irreducible quintic with rational coefficients.
+    Return an empty list if the quintic is reducible or not solvable.
     """
     result = []
-    coeff_5, coeff_4, p, q, r, s = f.all_coeffs()
-
-    # Eqn must be of the form x^5 + px^3 + qx^2 + rx + s
-    if coeff_4:
+    if not f.is_univariate:
         return result
 
+    coeff_5, coeff_4, p_, q_, r_, s_ = f.all_coeffs()
+
     if coeff_5 != 1:
-        l = [p/coeff_5, q/coeff_5, r/coeff_5, s/coeff_5]
-        if not all(coeff.is_Rational for coeff in l):
-            return result
-        f = Poly(f/coeff_5)
-    elif not all(coeff.is_Rational for coeff in (p, q, r, s)):
+        f = Poly(f / coeff_5, extension = True)
+        _, coeff_4, p_, q_, r_, s_ = f.all_coeffs()
```

Comment:
I understand setting `extension=True` is meant to handle cases where the coeffs are irrational, but here again I feel we need to clarify the API. The docstring currently says `f` is expected to have rational coefficients. Does this method in fact work when the coeffs are irrational? (Asking because I don't know.) As used by the `roots()` function, do we know `f` will always have rational coeffs by the time it gets here?

Reply:
I suppose that it could also handle cases where the quintic is a multiple of a rational one, say `sqrt(2)*(x**5-2)`, though not explicitly informed in the docstring. But maybe it seems a little redundant. If this should be removed, then one should check whether the coeffs are rational here before calling `f.all_coeffs()` since there is a test passing in irrational quintics.

Reply:
Ah, I see. You're referring to `solvers/tests/test_solvers.py:test_quintics_3`, which shows we actually do expect to sometimes receive polys `f` with irrational coeffs here. This seems reasonable then. But I would add a comment for the sake of future devs, explaining that this allows polys of the form you mention (i.e. constant multiples of rational ones) to pass.